compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -325,7 +319,51 @@ trait QuotesAndSplices {
         new TreeTypeMap(typeMap = typeMap).transform(shape1)
       }
 
-    (typeBindings, shape2, patterns)
+    (bindSymMapping, shape2, patterns)
+  }
+
+  /** For each type variable defined in the quote pattern we generate an equivalent
+   *  binding that will be as type variable in the encoded `unapply` of the quote pattern.
+   *
+   *  @return Mapping from type variable symbols defined in the quote pattern into
+   *          type variable `Bind` definitions for the `unapply` of the quote pattern.
+   *          This mapping retains the original type variable definition order.
+   */
+  private def unapplyBindingsMapping(quoted: Tree)(using Context): collection.Map[Symbol, Bind] = {
+    val mapping = mutable.LinkedHashMap.empty[Symbol, Symbol]
```

Comment:
It looks like we never actually query this map, we only call mapping.keys and mapping.values, could we replace it by directly defining oldBindings and newBindingsRefs as ListBuffers?

Reply:
There is a `mapping.contains(sym)`. I will simplify it to a `LinkedHashSet` and a `ListBuffer`.